guardian/typerighter
# Add creation and update meta to rule model

**NB:** based on https://github.com/guardian/typerighter/pull/306 – please review that first!

## What does this change?

Adds `createdAt`, `createdBy`, `updatedAt` and `updatedBy` to the rule model. They're updated automatically as rules are created and modified.

### This happens in the application logic! Why not write a database trigger to update `updatedAt`?

We could! But we'd still have to write the user data – it made some sense to me to keep all of the logic for our meta in one domain. If there's a preference for a DB trigger, we could do that instead. One advantage would be that there's no possibility of us missing an update as a result of an error in application code.

## How to test

- The automated tests should pass, and you should be convinced they cover both creation and updating cases.
- Try creating and updating rules in Postman, following the instructions in #305. When you create or update rules, are creation/update values correct? E.g. with Postman:

<img width="516" alt="Screenshot 2023-04-18 at 18 12 36" src="https://user-images.githubusercontent.com/7767575/232857682-505a70be-cacc-41f3-a877-e06bda3cce7a.png">
